x-pack/legacy/plugins/code/public/components/integrations/code_flyout_main.tsx

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, ReactNode } from 'react';
+import {
+  EuiTitle,
+  EuiFlyoutHeader,
+  EuiText,
+  EuiTabs,
+  EuiTab,
+  EuiIcon,
+  EuiLink,
+} from '@elastic/eui';
+import { CodeViewer } from './code_viewer';
+import { History } from './history';
+
+enum Tab {
+  code,
+  history,
+  blame,
+}
+
+export const CodeFlyoutMain = (props: { repo: string; file: string; revision: string }) => {
+  const [selectedTab, setSelectedTab] = React.useState<Tab>(Tab.code);
+
+  let content: ReactNode;
+
+  switch (selectedTab) {
+    case Tab.blame:
+      content = (
+        <CodeViewer
+          key="blame"
+          repo={props.repo}
+          file={props.file}
+          revision={props.revision}
+          showBlame={true}
+        />
+      );
+      break;
+    case Tab.history:
+      content = <History revision={props.revision} repo={props.repo} file={props.file} />;
+      break;
+    case Tab.code:
+      content = (
+        <CodeViewer key="code" repo={props.repo} file={props.file} revision={props.revision} />
+      );
+      break;
+  }
+
+  return (
+    <Fragment>
```

Comment:
Some high level questions, are we going to expose this `CodeFlyoutMain` for external sharing or just the `CodeViewer`?  I am thinking of moving the `Blame/History/Code` tabs into the component that we are going to share. But If you are going to expose the entire `CodeFlyoutMain`, then I think it's a bit too much because essentially we are going to share a Code Viewer component which can be mounted in anywhere, either a Flyout or any pages.

Reply:
I am suggesting to have some refactoring here to let the APM team have a `CodeSourceViewer` component with a clear defined property.